clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -141,7 +141,8 @@ public synchronized void incrementNodesTriedSinceLastSuccessfulRefresh() {
      * has been set by receiving stale metadata from a different cluster
      */
     public synchronized boolean shouldUpdateClusterMetadataFromBootstrap(long nowMs) {
-        return (this.nodesTriedSinceLastSuccessfulRefresh >= 1 &&
+        return this.maxClusterMetadataExpireTimeMs > 0 &&
```

Comment:
lgtm. Seems CC is using Metadata directly instead of MetadataUpdater, is that the reason it cannot override this method to false always?